### PR TITLE
PHOENIX-7316 Need close more Statements

### PR DESCRIPTION
**Pull request by chaijunjie0101**

### CODE REVIEW

File: phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java [hunk 1]
```diff
@@ -1342,9 +1342,11 @@ public ExecutableShowTablesStatement(String schema, String pattern) {
         @Override
         public QueryPlan compilePlan(final PhoenixStatement stmt, Sequence.ValueOp seqAction)
             throws SQLException {
-            PreparedStatement delegateStmt = QueryUtil.getTablesStmt(stmt.getConnection(), null,
-                getTargetSchema(), getDbPattern(), null);
-            return ((PhoenixPreparedStatement) delegateStmt).compileQuery();
+            try (PreparedStatement delegateStmt =
```
Comment (stoty) [inline phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java:1345]: This is suspicious.
We keep using the query plan returned from the closed statement.

File: phoenix-core-client/src/main/java/org/apache/phoenix/trace/PhoenixMetricsSink.java [hunk 2]
```diff
@@ -304,6 +306,14 @@ public void putMetrics(MetricsRecord record) {
         } catch (SQLException e) {
             LOGGER.error("Could not write metric: \n" + record + " to prepared statement:\n" + stmt,
                     e);
+        } finally {
```
Comment (stoty) [inline phoenix-core-client/src/main/java/org/apache/phoenix/trace/PhoenixMetricsSink.java:309]: Why not a simple try-with-resources ?

File: phoenix-core-client/src/main/java/org/apache/phoenix/trace/TraceWriter.java [hunk 3]
```diff
@@ -238,6 +239,14 @@ private void addToBatch(Span span) {
             } catch (SQLException e) {
                 LOGGER.error("Could not write metric: \n" + span + " to prepared statement:\n" + stmt,
                     e);
+            } finally {
```
Comment (stoty) [inline phoenix-core-client/src/main/java/org/apache/phoenix/trace/TraceWriter.java:242]: Why not simple try-with-resources?

File: phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java (hunk 1 above)
Comment (chaijunjie0101) [reply phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java]: thanks reviewing， rollbacked it.

File: phoenix-core-client/src/main/java/org/apache/phoenix/trace/PhoenixMetricsSink.java (hunk 2 above)
Comment (chaijunjie0101) [reply phoenix-core-client/src/main/java/org/apache/phoenix/trace/PhoenixMetricsSink.java]: changed it to try-with-resources

File: phoenix-core-client/src/main/java/org/apache/phoenix/trace/TraceWriter.java (hunk 3 above)
Comment (chaijunjie0101) [reply phoenix-core-client/src/main/java/org/apache/phoenix/trace/TraceWriter.java]: changed it to try-with-resources

File: phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java (hunk 1 above)
Comment (chaijunjie0101) [reply phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java]: > This is suspicious. We keep using the query plan returned from the closed statement.

thanks reviewing， rollbacked it.

File: phoenix-core-client/src/main/java/org/apache/phoenix/trace/PhoenixMetricsSink.java (hunk 2 above)
Comment (chaijunjie0101) [reply phoenix-core-client/src/main/java/org/apache/phoenix/trace/PhoenixMetricsSink.java]: > Why not a simple try-with-resources ?

changed it.

File: phoenix-core-client/src/main/java/org/apache/phoenix/trace/TraceWriter.java (hunk 3 above)
Comment (chaijunjie0101) [reply phoenix-core-client/src/main/java/org/apache/phoenix/trace/TraceWriter.java]: changed it.

> Why not simple try-with-resources?

changed it.

**Pull request closed without merging**